xapian-letor/api/featurelist_internal.h

```diff
@@ -0,0 +1,153 @@
+     */
+    void compute_termfreq();
+
+    /** This method calculates the inverse document frequency(idf) of query terms in the database.
+     *  This information is stored in inverse_doc_freq.
+     *
+     *  Note: idf of a term 't' is calculated as below:
+     *
+     *  idf(t) = log(N/df(t))
+     *                                  Where,
+     *                                  N = Total number of documents in database and
+     *                                  df(t) = number of documents containing term 't'
+     */
+    void compute_inverse_doc_freq();
+
+    /** This method calculates the length of the documents as number of 'terms'.
+     *  It calculates the length for three different parts: title, body and whole document.
+     *  This information is stored in doc_length in the following format.
+     *
+     *  @code
+     *  map<string, long int> len;
+     *  len["title"];
+     *  len["body"];
+     *  len["whole"];
+     *  @endcode
+     */
+    void compute_doc_length();
+
+    /** This method calculates the length of the collection in number of terms for different parts like 'title', 'body' and 'whole'.
+     *  This is calculated as a stored user metadata in omindex otherwise it is calculated out of scratch
+     *  (this might take some time depending upon the size of the database).
+     *  Length information is stored in collection_length in the following format.
+     *
+     *  @code
+     *  map<string, long int> len;
+     *  len["title"];
+     *  len["body"];
+     *  len["whole"];
+     *  @endcode
+     *
+     */
+    void compute_collection_length();
+
+    /** This method calculates the frequency of query terms in the whole database.
+     *  The information is stored in collection_termfreq in and used during the feature calculation methods.s
+     */
+    void compute_collection_termfreq();
+
+    /// Specify the database to use for feature building. This will be used by the Internal class.
+    void set_database(const Xapian::Database & db);
+
+    /** Specify the query to use for feature building. This will be used by the Internal class.
+     * @param query  Xapian::Query which has to be queried
+     * @exception Xapian::InvalidArgumentError will be thrown if an empty
+     * query is supplied
+     */
+    void set_query(const Xapian::Query & query);
+
+    /// Specify the document to use for feature building. This will be used by the Internal class.
+    void set_doc(const Xapian::Document & doc);
+
+    public:
+
+    /// Vector containing Feature pointer objects. Each will be used to return feature value.
+    std::vector<Feature *> feature;
+
+    /// This method computes all the statistics and stores them in their corresponding variables
+    void compute_statistics(const Xapian::Query & query, const Xapian::Database & db, const Xapian::Document & doc);
+
+    /// Populates the statistics needed by a Feature
+    void populate_feature(Feature *feature_);
+
```

Comment:
Unwanted blank line. I specifically asked you to look for these, so I'll stop here until you've had a chance to complete your improvements based on the last round of review.